prov/gni/src/gnix_atomic.c

```diff
@@ -662,6 +662,9 @@ ssize_t _gnix_atomic(struct gnix_fid_ep *ep,
 
 	COND_RELEASE(ep->requires_lock, &ep->vc_lock);
 
+	/* If a new VC was allocated, progress CM before returning. */
```

Comment:
Yes.  I think that is more clear.  It's really if the vc is not connected, rather that if it's a new vc.